app/services/failed_imports/creation_service.rb

```diff
@@ -0,0 +1,15 @@
+module FailedImports
+  class CreationService
+    AlreadyExistsError = Class.new(StandardError)
+    class << self
+      def call(params)
+        raise AlreadyExistsError if FailedImport.find_by(id: params[:id])
+
+        FailedImport.transaction do
+          failed_import = FailedImport.create!(id: params[:id], provider_id: params[:provider_id])
```

Comment:
No need to capture this in a variable if it's just being returned

Reply:
fair, I can refactor